plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestIcebergNessieCatalogConnectorSmokeTest.java

```diff
@@ -243,25 +266,110 @@ public void testRepeatUnregisterTable()
     @Test
     @Override
     public void testDropTableWithMissingMetadataFile()
+            throws Exception
     {
-        assertThatThrownBy(super::testDropTableWithMissingMetadataFile)
-                .hasMessageMatching("metadata location for register_table is not supported");
+        String tableName = "test_drop_table_with_missing_metadata_file_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 x, 'INDIA' y", 1);
+
+        Location metadataLocation = Location.of(getMetadataLocation(tableName));
+        Location tableLocation = Location.of(getTableLocation(tableName));
+
+        // Delete current metadata file
+        fileSystem.deleteFile(metadataLocation);
+        assertThat(fileSystem.newInputFile(metadataLocation).exists())
+                .describedAs("Current metadata file should not exist")
+                .isFalse();
+
+        // try to drop table
+        assertUpdate("DROP TABLE " + tableName);
+        assertThat(getQueryRunner().tableExists(getSession(), tableName)).isFalse();
+        assertThat(fileSystem.listFiles(tableLocation).hasNext())
+                .describedAs("Table location should exist")
+                .isTrue();
```

Comment:
The base tests assume table location to NOT exist. Hence, had to override as per Nessie bhaviour.